ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/RequestDataStep.java

```diff
@@ -129,7 +129,24 @@ public CompletableFuture<List<Task<SnapDataRequest>>> requestStorage(
             (response, error) -> {
               if (response != null) {
                 downloadState.removeOutstandingTask(getStorageRangeTask);
-                for (int i = 0; i < response.slots().size(); i++) {
+                final ArrayDeque<TreeMap<Bytes32, Bytes>> slots = new ArrayDeque<>();
+                // Check if we have an empty range
+
+                /*
+                 * Checks if the response represents an "empty range".
+                 *
+                 * An "empty range" is defined as a response where either no slots are present,
+                 * or the first slot is empty, but at least one proof exists
+                 */
+                final boolean isEmptyRange =
+                    (response.slots().isEmpty() || response.slots().get(0).isEmpty())
+                        && !response.proofs().isEmpty();
+                if (isEmptyRange) { // empty range detected
+                  slots.add(new TreeMap<>());
+                } else {
+                  slots.addAll(response.slots());
+                }
+                for (int i = 0; i < slots.size(); i++) {
```

Comment:
if the range is empty, won't slots.size() be 0 and we will never enter this loop, and never add a response to the request?

Reply:
It cannot be empty thanks to that https://github.com/hyperledger/besu/blob/8587a01be6f9d1b6a2aa7fbdca2bd2ba0a45fab9/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/RequestDataStep.java#L145